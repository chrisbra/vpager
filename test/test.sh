#!/bin/sh

#set -x

# Setup some color variables
red=$(tput setaf 1)
green=$(tput setaf 2)
reset=$(tput sgr0)

for i in */; do
  (
    cd "$i" &&
    test -f "cmd" &&
    ./cmd > test.out &&
    diff -qu test.expected  test.out >/dev/null ;
    if [ $? -ne 0 ]; then
      printf "Test %s:\t\t[${red}Failed${reset}]\n" ${i%%/}
      diff -u test.expected test.out
    else
      printf "Test %s:\t\t[${green}OK${reset}]\n" ${i%%/}
    fi
    )
done
