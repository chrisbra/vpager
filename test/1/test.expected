NAME

    This script vpager can be used as pager from inside Vims built-in
    terminal. The output will be copied to a new buffer in Vim.

SYNOPSIS
    vpager [-n -C option]
    vpager [-h|-v]

    -v	display version
    -h	display help
    -C	Pass options to Vim. Can be used to e.g. set the filetype for
        correct syntax highlighting.
    -n  Clear the previous buffer in Vim
    -V  Open a vertical split window in Vim

EXAMPLES
  git diff | vpager -nC 'ft=diff'

  Copies the output of git diff into a buffer inside Vim. Any previous
  output in the buffer will be cleared and the filetype will be set
  to "diff", for proper syntax highlighting.
